### PR TITLE
Cais doc changes

### PR DESCRIPTION
**Pull request by CaisManai**
### Why is this change needed?

This change to the docs is help engage and convert developers landing on the Obscuro developer docs page.

https://github.com/obscuronet/obscuro-internal/issues/949
https://github.com/obscuronet/obscuro-internal/issues/1040

### What changes were made as part of this PR:

Restructuring of the docs site and addition of a tutorial based on the Guessing Game

### CODE REVIEW

File: docs/testnet/essentials.md [hunk 1]
```diff
@@ -1,22 +1,20 @@
-# Essential Testnet Information
-An easy-to-read list of essential parameters and configuration settings for Obscuro Testnet.
+# Essential Information
+Essential information, parameters and configuration settings for the Oikos testnet.
+
+## Limitations on Oikos
```
Comment (GavT) [inline docs/testnet/essentials.md:4]: Oikos again!

File: docs/_data/navigation.yml [hunk 2]
```diff
@@ -1,3 +1,14 @@
+main:
+  - title: "Whitepaper"
+    url: testnet/Introduction.html
```
Comment (GavT) [inline docs/_data/navigation.yml:3]: These URLs are wrong for the title.

File: docs/testnet/essentials.md [hunk 3]
```diff
@@ -1,22 +1,20 @@
...
+These are the key limitations to be aware of when developing for the Oikos testnet:
+
+1. New nodes cannot be added. For now the number of Obscuro nodes is fixed.
+1. Data revelation is not implemented yet.
+1. Security is not fully implemented. Some keys are still hardcoded.
+1. The decentralised bridge is limited to two hardcoded ERC20 tokens.
+1. The Layer 1 is currently a hosted network. In the next iteration we'll connect Obscuro to Goerli.
+1. The "Wallet Extension" is not fully polished yet.
```
Comment (GavT) [inline docs/testnet/essentials.md:12]: What does that mean in reality? Expect a better UX in future?

File: docs/testnet/tutorial.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+- Use Obscuro concepts to develop a dApp
+
+What is the *Guessing Game*?
+
+The Guessing Game is a dApp that you will build that demonstrates a basic Obscuro use case, a simple number guessing game. The contract generates a random secret number, which is never revealed to anyone, including node operators, developers or end-user because of the privacy provided by Obscuro. The goal of the game is to guess this number, and each time an attempt is made, an entrance fee of 1 token is paid. If a user correctly guesses the number, the contract will pay out all of the accumulated entrance fees to them and reset itself with a new random number.
+
+Without Obscuro, it would be possible to look up the internal state of the contract and cheat, and the game wouldn't work.
+
+The dApp has many of the features that you'd expect to find, including:
+- Generating a random number known only to the dApp
```
Comment (GavT) [inline docs/testnet/tutorial.md:16]: add full stops?

File: docs/testnet/tutorial.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+- Allow users to guess the numbers by depositing a token for each play
+- Colder/Warmer functionality to help guide users to the correct number
+- Events to alert users whether they have won or not
+- Track all user's plays
```
Comment (GavT) [inline docs/testnet/tutorial.md:20]: plays or guesses?

File: docs/testnet/tutorial.md [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+![Step 2 screenshot](https://images.tango.us/workflows/57863807-6b96-4295-a7a1-42da1623f030/steps/4b3b212d-2518-4cc9-aab2-4befe128b38c/66a9a87d-0899-42d2-8f6a-98f204b25825.png?crop=focalpoint&fit=crop&fp-x=0.5000&fp-y=0.5000&fp-z=1.0000&w=1200&mark-w=0.2&mark-pad=0&mark64=aHR0cHM6Ly9pbWFnZXMudGFuZ28udXMvc3RhdGljL21hZGUtd2l0aC10YW5nby13YXRlcm1hcmsucG5n&ar=530%3A371)
+
+
+#### Select Guess.sol, and we're ready
+![Step 1 screenshot](https://images.tango.us/workflows/bc1e6526-eb4e-4815-8690-412c37c1c5cb/steps/3a33b699-a3a9-42b5-b33a-2826f4516b08/faa2c399-1f01-4263-90d4-5d31f075e9bd.png?crop=focalpoint&fit=crop&fp-x=0.5000&fp-y=0.5000&fp-z=1.0000&w=1200&mark-w=0.2&mark-pad=0&mark64=aHR0cHM6Ly9pbWFnZXMudGFuZ28udXMvc3RhdGljL21hZGUtd2l0aC10YW5nby13YXRlcm1hcmsucG5n&ar=591%3A175)
+
+
+# Let's code
+Guess.sol is a Solidity file, and once we're done, it will contain everything we need for the Guessing Game. Inside the file you'll find the classic ERC-20 interface and a simple implementation named ERC20Basic representing OGG (Obscuro Guessing Game token). We'll use OGG as means of players entering to play and the prize pool.
+
```
Comment (GavT) [inline docs/testnet/tutorial.md:47]: We'll use OGG as a means of entry to play and to generate the prize pool.

File: docs/testnet/tutorial.md [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+![Step 1 screenshot](https://images.tango.us/workflows/41b454a9-b560-4d60-8775-3e40c5bc1466/steps/74347d9b-e3fe-48aa-8e74-fecebd22fdfa/80231e65-e0be-4bf2-8150-521721745b2c.png?crop=focalpoint&fit=crop&fp-x=0.3145&fp-y=0.5909&fp-z=3.3270&w=1200&mark-w=0.2&mark-pad=0&mark64=aHR0cHM6Ly9pbWFnZXMudGFuZ28udXMvc3RhdGljL21hZGUtd2l0aC10YW5nby13YXRlcm1hcmsucG5n&ar=2307%3A1244)
+
+Let's start by defining the key variables we'll need for the game. One is to store the secret number that only the contract knows about, and another containing the ERC-20 contract.
+
+```
+    uint8 private _target
+    IERC20 public erc20
+```
+Upon seeing this, you might think, hang on, I can _target is set to private, but so what? Private variables in Ethereum don't really mean they're private, just that they can't be accessed or modified by other smart contracts. And you're right. That's the case in Ethereum. But in Obscuro, private variables *really* are private (unless you explicitly decide to reveal them!). This is because all contracts and contract data is encrypted and only accessible within enclaves.
```
Comment (GavT) [inline docs/testnet/tutorial.md:58]: I can... see? Should _target be in code? `_target`

Comment (GavT) [inline docs/testnet/tutorial.md:58]: And you're right, that's the case in Ethereum.

File: docs/testnet/tutorial.md [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+- a constructor that initializes the contract
+- a way to calculate the random number to be guessed
+
+We now need to add the ability for a player to make an actual guess. But before we do that, we have not thought about the range we might want numbers to appear in. This would allow us to set the difficulty for any particular game. So let's do that now.
+
+Extend the constructor to accept a range int and create a variable for it, so we have something that looks like this
+
+![Step 1 screenshot](https://images.tango.us/workflows/8f7ef3ef-091f-4b5d-9017-d041449139ae/steps/43aff627-6025-42c1-8e1a-654761d01b90/c3fc2de4-e9fa-4d31-b1a8-abe5331e0ec9.png?crop=focalpoint&fit=crop&fp-x=0.2957&fp-y=0.3023&fp-z=4.0000&w=1200&mark-w=0.2&mark-pad=0&mark64=aHR0cHM6Ly9pbWFnZXMudGFuZ28udXMvc3RhdGljL21hZGUtd2l0aC10YW5nby13YXRlcm1hcmsucG5n&ar=2307%3A1244)
+
+And we can change our random number function to adhere to the range by including the modulo operator %, like this:
```
Comment (GavT) [inline docs/testnet/tutorial.md:91]: modulo?

File: docs/testnet/tutorial.md [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+Next, let's add code that checks if the guess is correct or not. If the guess is correct, call the ERC-20 function to transfer all the funds in the contract and reset the game by calling _setNewTarget().
+
+If it's not correct, we do nothing. For example, your code might look something like this:
+
+![Step 1 screenshot](https://images.tango.us/workflows/7073e589-7d79-491f-b4f1-27db2eb144c1/steps/ad4b9d9a-5793-420f-b21c-3a22ae8e83b5/b016b98e-203a-4362-9c14-6ff2928b9ba8.png?crop=focalpoint&fit=crop&fp-x=0.5000&fp-y=0.5000&fp-z=1.0000&w=1200&mark-w=0.2&mark-pad=0&mark64=aHR0cHM6Ly9pbWFnZXMudGFuZ28udXMvc3RhdGljL21hZGUtd2l0aC10YW5nby13YXRlcm1hcmsucG5n&ar=879%3A246)
+
+
+This works. You now have an Obscuro contract that will let players play the guessing game. But it's not very good. So how do players know if they've won or not? We need events!
+
+In Ethereum, events are typically written to the event log that is viewable by everyone. In Obscuro, events are viewable only to those they're intended for. This is done by encrypting events with a key that is known only to the intended recipient. This is handled automatimatically for the recipient through the Wallet Extension.
```
Comment (GavT) [inline docs/testnet/tutorial.md:128]: automiatacialsfmslemgly typo :-)

File: docs/testnet/tutorial.md [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+```
+
+Where **prize** represents the current prize pool so we can remind the player of what they're playing for.
+
+With these events in place, we can now call them accordingly within our attempt function, so we end up with something like this:
+
+![Step 1 screenshot](https://images.tango.us/workflows/263b9f82-3d6e-46d1-908f-952591a206d8/steps/cb679525-fb5f-4cae-991a-02285beee941/765986b3-1999-4814-a6ac-89bb6d22488d.png?crop=focalpoint&fit=crop&fp-x=0.5000&fp-y=0.5000&fp-z=1.0000&w=1200&mark-w=0.2&mark-pad=0&mark64=aHR0cHM6Ly9pbWFnZXMudGFuZ28udXMvc3RhdGljL21hZGUtd2l0aC10YW5nby13YXRlcm1hcmsucG5n&ar=870%3A269)
+
+
+Before we move on, we've seen how we can keep things private (hint: all of the above is private!). What about if there are things we want to intentionally make public? Foe example, it makes sense for the current prize pool to be public, so let's go ahead and explore that. It's straightforward. All you do is create a public view that returns something - precisely the same as any other EVM contract!
```
Comment (GavT) [inline docs/testnet/tutorial.md:146]: foe >> for (Typo)

File: docs/_data/navigation.yml [hunk 11]
```diff
@@ -7,36 +18,38 @@ pages:
         url: what-is-obscuro/technology.html
       - title: User Experience
         url: what-is-obscuro/user-experience.html
-      - title: Tokens
-        url: what-is-obscuro/tokens.html
-      - title: Community
-        url: what-is-obscuro/community.html
+      #- title: Community
```
Comment (joeldudley) [inline docs/_data/navigation.yml:21]: Shouldn't you get rid of all these headers, instead of commenting them out?

File: docs/index.md [hunk 12]
```diff
@@ -1,7 +1,7 @@
-<p style="text-align: center;">
-    <img src="assets\images\Obscuro-white-strip.png">
+<p style="text-align: left;" class="build">
+    <img src="assets\images\buildwithprivacy.png">
 </p>
 
-Obscuro is a decentralised Ethereum Layer 2 Rollup protocol with privacy.
+Welcome to Obscuro. Privacy for Ethereum on an L2 roll-up. No SDKs, 100% EVM.
```
Comment (joeldudley) [inline docs/index.md:5]: Is "An L2 for privacy on Ethereum" better?

And are we assuming too much to assume that people know what an L2 is? Should we spell out "layer 2"?

File: docs/index.md [hunk 13]
```diff
@@ -1,7 +1,7 @@
...
 
 On this docsite you will find useful guidance on Obscuro, how to participate with the Testnet and, if you want to go deeper, read the whitepaper using the menu on the left. A PDF version of the whitepaper is [available](assets/images/obscuro-whitepaper-0-10-0.pdf).
```
Comment (joeldudley) [inline docs/index.md:7]: Not your change, but should be "participate in the Testnet".

File: docs/testnet/Introduction.md [hunk 14]
```diff
@@ -1,16 +1,12 @@
-# Introducing Obscuro Testnet
-The Obscuro Testnet is a public network used by developers, node operators and end users to test Obscuro before deploying to Mainnet.
+# Introducing Oikos
```
Comment (joeldudley) [inline docs/testnet/Introduction.md:1]: I'm with Gavin. Somewhere here, you have to say that Oikos is a testnet.

File: docs/testnet/Introduction.md [hunk 15]
```diff
@@ -1,16 +1,12 @@
...
+Oikos can also be used by end users to help test Obscuro and test developer dApps. Oikos does not focus on node operators. Support for node operators will arrive in the next version of testnet. 
+
+Developers can expect frequent updates which will be communicated [here](https://docs.obscu.ro/testnet/changelog.html). The trade-off is that in the early days Testnet will have some sharp edges and unexpected surprises. More specifically, whilst Testnet is in its infancy:
 * Obscuro Testnet will crash from time to time, and will frequently be updated with new features.
 * When Obscuro Testnet restarts expect to lose everything.
 * Obscuro Testnet code has not been optimised or audited yet. (Mainnet code will be audited prior to launch.)
 * The [Wallet extension](https://docs.obscu.ro/wallet-extension/wallet-extension.html) is still very experimental.
```
Comment (joeldudley) [inline docs/testnet/Introduction.md:10]: I wouldn't call out the WE as particularly experimental?

File: docs/testnet/Introduction.md [hunk 16]
```diff
@@ -1,16 +1,12 @@
...
+# Introducing Oikos
+This section contains detailed on how to connect, build and deploy on Oikos. Oikos is designed with developers in mind. It provides the latest upgrades and tooling for developers to deploy and test their dApps.
 
-The Testnet release schedule has been intentionally designed to make new features available as soon as possible. This means you can expect frequent updates which will be communicated [here](https://docs.obscu.ro/testnet/changelog.html). The trade-off is that in the early days Testnet will have some sharp edges and unexpected surprises. More specifically, whilst Testnet is in its infancy:
+Oikos can also be used by end users to help test Obscuro and test developer dApps. Oikos does not focus on node operators. Support for node operators will arrive in the next version of testnet. 
+
+Developers can expect frequent updates which will be communicated [here](https://docs.obscu.ro/testnet/changelog.html). The trade-off is that in the early days Testnet will have some sharp edges and unexpected surprises. More specifically, whilst Testnet is in its infancy:
 * Obscuro Testnet will crash from time to time, and will frequently be updated with new features.
```
Comment (joeldudley) [inline docs/testnet/Introduction.md:7]: Seems odd to have "new features" listed in a list of negatives.

File: docs/testnet/essentials.md [hunk 17]
```diff
@@ -1,22 +1,20 @@
-# Essential Testnet Information
-An easy-to-read list of essential parameters and configuration settings for Obscuro Testnet.
+# Essential Information
+Essential information, parameters and configuration settings for the Oikos testnet.
+
+## Limitations on Oikos
+These are the key limitations to be aware of when developing for the Oikos testnet:
```
Comment (joeldudley) [inline docs/testnet/essentials.md:5]: A lot of these limitations wouldn't be limitations related to _developing_ for testnet. Maybe just say, "key limitations to be aware of when using the Oikos testnet".

File: docs/testnet/essentials.md [hunk 18]
```diff
@@ -1,22 +1,20 @@
...
+1. Security is not fully implemented. Some keys are still hardcoded.
+1. The decentralised bridge is limited to two hardcoded ERC20 tokens.
+1. The Layer 1 is currently a hosted network. In the next iteration we'll connect Obscuro to Goerli.
+1. The "Wallet Extension" is not fully polished yet.
 
 ## Connection to an Obscuro Node
 - **RPC http address:** `testnet.obscu.ro:13000`
 - **RPC websocket address:** `testnet.obscu.ro:13001`
 
-## Custom Network for MetaMask
```
Comment (joeldudley) [inline docs/testnet/essentials.md:8]: How come we're removing this and the ObscuroScan URL?

File: docs/testnet/tutorial.md [hunk 19]
```diff
@@ -0,0 +1,158 @@
...
+
+The dApp has many of the features that you'd expect to find, including:
+- Generating a random number known only to the dApp
+- Allow users to guess the numbers by depositing a token for each play
+- Colder/Warmer functionality to help guide users to the correct number
+- Events to alert users whether they have won or not
+- Track all user's plays
+- Contract owner privileges to start and stop the game
+
+# Set up your environment
```
Comment (joeldudley) [inline docs/testnet/tutorial.md:23]: I'll try out the tutorial tomorrow.

I was wondering if we could just take an existing Ethereum contract (ERC20 or something) and add privacy to that, to make a simpler demo?

File: docs/testnet/Introduction.md (hunk 14 above)
Comment (GavT) [reply docs/testnet/Introduction.md:1]: Can we choose a better name that is easier to pronounce? We have the entire universe of language at our disposal so lets make its something easy (Oi-kos? o-i-kos? oik-os?)

File: docs/testnet/tutorial.md (hunk 8 above)
Comment (CaisManai) [reply docs/testnet/tutorial.md:91]: It gives you the remainder when one number is divided by another

File: docs/testnet/Introduction.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (CaisManai) [reply docs/testnet/Introduction.md:1]: One of the top 5 names chosen for testnet. A place holder until we make a final decision

File: docs/_data/navigation.yml (hunk 2 above)
Comment (CaisManai) [reply docs/_data/navigation.yml:3]: doh

File: docs/testnet/Introduction.md (hunk 14 above)
Comment (joeldudley) [reply docs/testnet/Introduction.md:1]: To make the pronunciation crystal clear, let's spell it `οἶκος`.

File: docs/testnet/tutorial.md [hunk 20]
```diff
@@ -0,0 +1,158 @@
+# Getting Started
+In this tutorial, you build your own Obscuro dApp from the start. This is an excellent way to experience a typical development process as you learn how Obscuro dApps are built, design concepts, tools and terminology.
```
Comment (joeldudley) [inline docs/testnet/tutorial.md:2]: Highlight here that you can also reuse existing apps with no changes?

File: docs/testnet/tutorial.md [hunk 21]
```diff
@@ -0,0 +1,158 @@
...
+
+# Let's go
+This 'Guessing Game' tutorial provides an introduction to the fundamentals of Obscuro and shows you how to:
```
Comment (joeldudley) [inline docs/testnet/tutorial.md:5]: I'd reword this as, "In this tutorial, we'll build a 'Guessing Game' dApp that provides..."

File: docs/testnet/tutorial.md [hunk 22]
```diff
@@ -0,0 +1,158 @@
...
+- Set up your local Ethereum development environment for Obscuro.
+- Use Obscuro concepts to develop a dApp
+
+What is the *Guessing Game*?
+
+The Guessing Game is a dApp that you will build that demonstrates a basic Obscuro use case, a simple number guessing game. The contract generates a random secret number, which is never revealed to anyone, including node operators, developers or end-user because of the privacy provided by Obscuro. The goal of the game is to guess this number, and each time an attempt is made, an entrance fee of 1 token is paid. If a user correctly guesses the number, the contract will pay out all of the accumulated entrance fees to them and reset itself with a new random number.
```
Comment (joeldudley) [inline docs/testnet/tutorial.md:11]: I think you can start strong here and say, "The Guessing Game is an example of a dApp that can only be built on Obscuro, and not on native Ethereum"

Comment (joeldudley) [inline docs/testnet/tutorial.md:11]: "The contract generates" -> "The dApp's smart contract generates"

File: docs/testnet/tutorial.md [hunk 23]
```diff
@@ -0,0 +1,158 @@
...
+
+Without Obscuro, it would be possible to look up the internal state of the contract and cheat, and the game wouldn't work.
```
Comment (joeldudley) [inline docs/testnet/tutorial.md:13]: Can you also highlight that you cannot generate random numbers in Ethereum smart contracts, and only in Obscuro? Or is that not true?

File: docs/testnet/tutorial.md [hunk 24]
```diff
@@ -0,0 +1,158 @@
...
+- Allow users to guess the numbers by depositing a token for each play
+- Colder/Warmer functionality to help guide users to the correct number
+- Events to alert users whether they have won or not
+- Track all user's plays
+- Contract owner privileges to start and stop the game
+
+# Set up your environment
+- You'll need to install MetaMask following the instructions [here](https://metamask.io/)
+- Then download and run the Obscuro Wallet extension following the instructions [here](https://docs.obscu.ro/wallet-extension/wallet-extension.html)
+- Now configure MetaMask following the instructions [here](https://docs.obscu.ro/wallet-extension/configure-metamask.html)
```
Comment (joeldudley) [inline docs/testnet/tutorial.md:26]: "Now configure MetaMask" -> "Now connect MetaMask to the Obscuro testnet"?

File: docs/testnet/tutorial.md [hunk 25]
```diff
@@ -0,0 +1,158 @@
...
+- Finally, check you can open the Remix IDE by visiting the following URL in your browser https://remix.ethereum.org
+
+That's it. You're all set to start building your first dApp on Obscuro.
+
+## 1. To begin, we'll clone a simple repo from Github through Remix by following these instructions
```
Comment (joeldudley) [inline docs/testnet/tutorial.md:31]: "we'll clone a simple repo" -> "we'll clone a template smart contract"?

File: docs/testnet/tutorial.md [hunk 26]
```diff
@@ -0,0 +1,158 @@
...
+
+#### Click on GitHub
```
Comment (joeldudley) [inline docs/testnet/tutorial.md:33]: Add a step before this to open Remix?

File: docs/testnet/tutorial.md [hunk 27]
```diff
@@ -0,0 +1,158 @@
...
+![Step 1 screenshot](https://images.tango.us/workflows/57863807-6b96-4295-a7a1-42da1623f030/steps/919b227f-f1f5-4af3-aba1-7fb83576d505/8144f3f5-7855-4c83-86dd-c88e5476c367.png?crop=focalpoint&fit=crop&fp-x=0.5000&fp-y=0.5000&fp-z=1.0000&w=1200&mark-w=0.2&mark-pad=0&mark64=aHR0cHM6Ly9pbWFnZXMudGFuZ28udXMvc3RhdGljL21hZGUtd2l0aC10YW5nby13YXRlcm1hcmsucG5n&ar=874%3A594)
+
+
+#### Paste "https://github.com/obscuronet/sample-applications/blob/main/number-guessing-game/contracts/Guess.sol" into input
+![Step 2 screenshot](https://images.tango.us/workflows/57863807-6b96-4295-a7a1-42da1623f030/steps/4b3b212d-2518-4cc9-aab2-4befe128b38c/66a9a87d-0899-42d2-8f6a-98f204b25825.png?crop=focalpoint&fit=crop&fp-x=0.5000&fp-y=0.5000&fp-z=1.0000&w=1200&mark-w=0.2&mark-pad=0&mark64=aHR0cHM6Ly9pbWFnZXMudGFuZ28udXMvc3RhdGljL21hZGUtd2l0aC10YW5nby13YXRlcm1hcmsucG5n&ar=530%3A371)
+
+
+#### Select Guess.sol, and we're ready
```
Comment (joeldudley) [inline docs/testnet/tutorial.md:41]: "Select Guess.sol to open it in the editor"

File: docs/testnet/tutorial.md [hunk 28]
```diff
@@ -0,0 +1,158 @@
...
+![Step 1 screenshot](https://images.tango.us/workflows/bc1e6526-eb4e-4815-8690-412c37c1c5cb/steps/3a33b699-a3a9-42b5-b33a-2826f4516b08/faa2c399-1f01-4263-90d4-5d31f075e9bd.png?crop=focalpoint&fit=crop&fp-x=0.5000&fp-y=0.5000&fp-z=1.0000&w=1200&mark-w=0.2&mark-pad=0&mark64=aHR0cHM6Ly9pbWFnZXMudGFuZ28udXMvc3RhdGljL21hZGUtd2l0aC10YW5nby13YXRlcm1hcmsucG5n&ar=591%3A175)
+
+
+# Let's code
+Guess.sol is a Solidity file, and once we're done, it will contain everything we need for the Guessing Game. Inside the file you'll find the classic ERC-20 interface and a simple implementation named ERC20Basic representing OGG (Obscuro Guessing Game token). We'll use OGG as means of players entering to play and the prize pool.
```
Comment (joeldudley) [inline docs/testnet/tutorial.md:46]: Would it be possible to have the `IERC20` and `ERC20` interface and contract defined in a separate file? Or at least moved to the bottom of the file? It's quite scary to open a template and see that much code.

File: docs/testnet/Introduction.md (hunk 14 above)
Comment (CaisManai) [reply docs/testnet/Introduction.md:1]: I don't really like Oikos TBH. So I'm with you guys, let's choose a better name

Comment (joeldudley) [reply docs/testnet/Introduction.md:1]: Evan's Cat

File: docs/testnet/tutorial.md (hunk 23 above)
Comment (CaisManai) [reply docs/testnet/tutorial.md:13]: I think you can now with Eth2 using the beacon chain

File: docs/testnet/tutorial.md (hunk 28 above)
Comment (CaisManai) [reply docs/testnet/tutorial.md:46]: Agreed

File: docs/testnet/tutorial.md (hunk 27 above)
[the remark this answers is not in the document]
Comment (CaisManai) [reply docs/testnet/tutorial.md:41]: Oops

File: docs/testnet/tutorial.md (hunk 19 above)
Comment (CaisManai) [reply docs/testnet/tutorial.md:23]: Good idea. I've added a ticket and will add that as another tutorial/sample

File: docs/testnet/essentials.md (hunk 18 above)
Comment (CaisManai) [reply docs/testnet/essentials.md:8]: They've been moved e.g. ObscuroScan is now accessed directly from the menu under tools

File: docs/testnet/tutorial.md (hunk 27 above)
Comment (joeldudley) [inline docs/testnet/tutorial.md:41]: I just tried it, and the file window didn't autoexpand to show the contract (it did the last time, weirdly). Worth adding a step to navigate to the file, just in case?

File: docs/testnet/tutorial.md (hunk 7 above)
Comment (joeldudley) [inline docs/testnet/tutorial.md:58]: This should be, "I can see _target".

Comment (joeldudley) [inline docs/testnet/tutorial.md:58]: I think you can get rid of the "and only accessible within enclaves". Stronger to just say it's encrypted.

File: docs/testnet/tutorial.md [hunk 29]
```diff
@@ -0,0 +1,158 @@
...
+
+There is also a skeleton contract called Guess inside Guess.sol, and this is what we'll be extending. The contract Guess will be written the same as any other Solidity contract in Ethereum. It'll run on the EVM and will exhibit the same behaviours.
+
+![Step 1 screenshot](https://images.tango.us/workflows/41b454a9-b560-4d60-8775-3e40c5bc1466/steps/74347d9b-e3fe-48aa-8e74-fecebd22fdfa/80231e65-e0be-4bf2-8150-521721745b2c.png?crop=focalpoint&fit=crop&fp-x=0.3145&fp-y=0.5909&fp-z=3.3270&w=1200&mark-w=0.2&mark-pad=0&mark64=aHR0cHM6Ly9pbWFnZXMudGFuZ28udXMvc3RhdGljL21hZGUtd2l0aC10YW5nby13YXRlcm1hcmsucG5n&ar=2307%3A1244)
+
+Let's start by defining the key variables we'll need for the game. One is to store the secret number that only the contract knows about, and another containing the ERC-20 contract.
```
Comment (joeldudley) [inline docs/testnet/tutorial.md:52]: It wasn't too clear to me what these variables meant. I wonder if we rename `_target` to `secretNumber` or similar?

And I think you should change "containing the ERC-20 contract" to "containing the ERC-20 contract to manage OGG tokens".

**Comment by joeldudley** [comment]
One general comment - I found it annoying that you couldn't copy and paste the code changes into Remix (because they're screenshots). But the screenshots are useful to see you've done it right. Hmm.

File: docs/testnet/tutorial.md [hunk 30]
```diff
@@ -0,0 +1,158 @@
...
+
+
+Let's extend to the constructor to include these:
+
+![Step 2 screenshot](https://images.tango.us/workflows/a8278591-dc09-4e26-97a2-b30776f86179/steps/58146527-d6d4-47c4-a796-cc0b3b4909db/56afa096-9f71-49a0-a85d-94ba3ecc56a4.png?crop=focalpoint&fit=crop&fp-x=0.5000&fp-y=0.5000&fp-z=1.0000&w=1200&mark-w=0.2&mark-pad=0&mark64=aHR0cHM6Ly9pbWFnZXMudGFuZ28udXMvc3RhdGljL21hZGUtd2l0aC10YW5nby13YXRlcm1hcmsucG5n&ar=410%3A117)
+
+
+Now let's write our function to generate the random number and secretly store it within the contract
+
+![Step 1 screenshot](https://images.tango.us/workflows/a8278591-dc09-4e26-97a2-b30776f86179/steps/8b18e51b-b4da-43e6-a752-eea0cc550791/009e7455-a298-4b66-808b-39080f30af24.png?crop=focalpoint&fit=crop&fp-x=0.5000&fp-y=0.5000&fp-z=1.0000&w=1200&mark-w=0.2&mark-pad=0&mark64=aHR0cHM6Ly9pbWFnZXMudGFuZ28udXMvc3RhdGljL21hZGUtd2l0aC10YW5nby13YXRlcm1hcmsucG5n&ar=864%3A211)
```
Comment (joeldudley) [inline docs/testnet/tutorial.md:76]: Typo in the screenshot - "Balanace"

File: docs/testnet/tutorial.md [hunk 31]
```diff
@@ -0,0 +1,158 @@
...
+
+
+This isn't great as although the number is arguably random, it's not random to anyone who knows how it's generated and how to measure the inputs when the function is called. We are working on something better. Watch this space!
+
+Okay, so right now, we should have the following:
+- a constructor that initializes the contract
+- a way to calculate the random number to be guessed
+
+We now need to add the ability for a player to make an actual guess. But before we do that, we have not thought about the range we might want numbers to appear in. This would allow us to set the difficulty for any particular game. So let's do that now.
```
Comment (joeldudley) [inline docs/testnet/tutorial.md:85]: I wonder if it would be cleaner to define guessRange back when you're updating the constructor earlier, then there would be one less step and you wouldn't have to come here and talk about modulo.

File: docs/testnet/tutorial.md [hunk 32]
```diff
@@ -0,0 +1,158 @@
...
+
+Before we move on, we've seen how we can keep things private (hint: all of the above is private!). What about if there are things we want to intentionally make public? Foe example, it makes sense for the current prize pool to be public, so let's go ahead and explore that. It's straightforward. All you do is create a public view that returns something - precisely the same as any other EVM contract!
+
+Our function to return the prize pool is:
+
+![Step 1 screenshot](https://images.tango.us/workflows/74ec7481-8c33-401f-8476-07c777e7cd3d/steps/d4a753b9-f507-4f6c-b080-40587ec82c00/8ffdc581-6e95-40bb-9c11-d3fa0ef01d84.png?crop=focalpoint&fit=crop&fp-x=0.5000&fp-y=0.5000&fp-z=1.0000&w=1200&mark-w=0.2&mark-pad=0&mark64=aHR0cHM6Ly9pbWFnZXMudGFuZ28udXMvc3RhdGljL21hZGUtd2l0aC10YW5nby13YXRlcm1hcmsucG5n&ar=518%3A87)
+
+And that's it! Congratulations on writing your first Obscuro smart contract!
+
+You can now go ahead and deploy the contract following the instructions [here](https://docs.obscu.ro/testnet/deploying-a-smart-contract.html)
```
Comment (joeldudley) [inline docs/testnet/tutorial.md:154]: I think it would be good to have information here on how to interact with the smart contract - e.g. how to make guesses.

**Review by joeldudley — DISMISSED** [review]
Just a few more suggestions.

In general, I worry this is the wrong tutorial - one of the marketing messages of Obscuro is that you don't need to change your contracts, so why are we writing one from scratch?

And then there's nothing in the tutorial actually showing to the user that the random number is encrypted.

I wonder if it would be better to start from an ERC20 contract or greeter, and modify that so that a single field is private in a demonstrable way.

**Review by GavT — APPROVED** [review]
LGTM
